docs/develop/smart-contracts/guidelines/message-delivery-guarantees.mdx

```diff
@@ -0,0 +1,70 @@
+
+## Message delivery
+
+Fortunately, TON is working such way that any internal message will surely be received on the destination account. Message cannot be lost anywhere between the source and destination. External messages is a little bit different since it's acceptance to the block is at validators discretionm however once message is accepted to the incoming message queue it will be delivered.
+
+### Delivery order
+
+It looks like _lt_ solves the issue about message delivery order, because we know that transaction with lower _lt_ will be processed first. But doesn't work for all scenarios.
+
+Suppose that there are two contracts - _A_ and _B_. Then _A_ receives an external message which triggers it to send two internal messages to _B_, let's call these messages _1_ and _2_. In that simple case we can be 100% sure that _1_ will be processed by _B_ before _2_ because it has lower _lt_.
+
+<ConceptImage src="/img/docs/msg-delivery-1.png" />
+
+It is just a simple case when we have only two contracts. What if our system works with more of them?
+
+### Several smart contracts
+
+Suppose that we have three contracts - _A_, _B_ and _C_. In some transaction, _A_ sends two internal messages _1_ and _2_: one to _B_ and another one to _C_. Even though they were created in some exact order (_1_, then _2_), we can't be sure that _1_ will be processed before _2_. That happens because routes from _A_ to _B_ and from _A_ to _C_ can differ in length and validator sets. If these contracts are in different shardchains, one of the messages may require several blocks to reach the destination contract.
+
+<ConceptImage src="/img/docs/msg-delivery-2.png" />
+
+The same thing happens in a reverse situation, when two contracts _B_ and _C_ send a message to one contract _A_. Even if message `B -> A` was sent before `C -> A`, we can't know which of them will be deliveried first. `B -> A` route may require more shardchain hops.
+
+<ConceptImage src="/img/docs/msg-delivery-3.png" />
+
+### Chains of messages
+
+Even more interesting things can happend when we have chains of messages. Let's have a look at a few examples.
+
+Suppose that we have contracts A, B and C. In some transcation, A sends two internal messages 1 and 2: one to B and one to C. And after receiving 1, contract B also sends a message 3 to C. In this case we can be 100% sure that message 2 will be deliveried before message 3, because the path `A -> C` is shorter than `A -> B -> C`.
```

Comment:
Suppose that we have contracts A, B and C. In **a** transaction, A sends two internal messages 1 and 2: one to B and one to C. And after receiving 1, contract B also sends a message 3 to C. In this case we can be 100% sure that message 2 will be **delivered** before message 3, because the path `A -> C` is shorter than `A -> B -> C`.